src/pages/index.js

```diff
@@ -0,0 +1,318 @@
+  likeButton.addEventListener("click", () => {
+    const isCurrentlyLiked = likeButton.classList.contains(
+      "card__like-button_active"
+    );
+    const apiCall = isCurrentlyLiked ? api.removeLike(_id) : api.addLike(_id);
+
+    apiCall
+      .then((updatedCard) => {
+        const newIsLiked = updatedCard.isLiked;
+        const newLikes =
+          parseInt(likeCount.textContent, 10) + (newIsLiked ? 1 : -1);
+
+        likeButton.classList.toggle("card__like-button_active", newIsLiked);
+        if (likeCount) likeCount.textContent = newLikes;
+      })
+      .catch((err) => console.error("Error updating like:", err));
+  });
+
+  deleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, _id)
+  );
+  cardImage.addEventListener("click", () => openPreviewModal(link, name));
+
+  return cardElement;
+};
+
+const openPreviewModal = (imageSrc, imageAlt) => {
+  selectors.previewImage.src = imageSrc;
+  selectors.previewImage.alt = imageAlt;
+  selectors.previewCaption.textContent = imageAlt;
+  openModal(selectors.previewModal);
+};
+
+const renderCard = (item, method = "prepend") => {
+  if (document.querySelector(`[data-id="${item._id}"]`)) {
+    console.log("Card already exists, skipping:", item._id);
+    return;
+  }
+  const card = createCardElement(item);
+  card.setAttribute("data-id", item._id); // Use data attribute to track the card
+  console.log("Card element created:", card);
+  selectors.cardsContainer[method](card);
+};
+
+// ------------------- EVENT LISTENERS -------------------
+const avatarEditButton = document.querySelector(".profile__avatar-btn");
+const avatarForm = document.querySelector("#edit-avatar-form");
+const deleteForm = document.querySelector("#delete-card-form");
+
+avatarEditButton.addEventListener("click", () =>
+  openModal(selectors.avatarModal)
+);
+avatarForm.addEventListener("submit", handleAvatarFormSubmit);
+selectors.profileForm.addEventListener("submit", handleProfileFormSubmit);
+selectors.cardForm.addEventListener("submit", handleCardFormSubmit);
+selectors.profileEditButton.addEventListener("click", () => {
+  selectors.nameInput.value = selectors.profileName.textContent;
+  selectors.jobInput.value = selectors.profileDescription.textContent;
+  openModal(selectors.editProfileModal);
+});
+selectors.cardAddButton.addEventListener("click", () =>
+  openModal(selectors.cardModal)
+);
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+// ------------------- MODAL CLOSE EVENTS -------------------
+document.addEventListener("DOMContentLoaded", () => {
+  document.querySelectorAll(".modal").forEach((modal) => {
+    const closeButton = modal.querySelector(".modal__close");
+    if (closeButton) {
+      closeButton.addEventListener("click", () => closeModal(modal));
+    }
+    modal.addEventListener("click", (event) => {
+      if (event.target === modal) closeModal(modal);
+    });
+  });
+
+  document.querySelectorAll(".modal__input").forEach((input) => {
```

Comment:
It's not needed because all validation is already inside `validation.js`. You shouldn't add it in `index.js` again